backend/src/AppServer.hs

```diff
@@ -40,14 +40,14 @@ processReport timestamp winner loser report =
       loserRatingAfter = 0 -- TODO
     }
 
-submitReportHandler :: GameReport -> AppM ()
+submitReportHandler :: GameReport -> AppM ReadProcessedGameReport
 submitReportHandler r = case validateReport r of
   Failure errors -> throwError $ err422 {errBody = show errors}
   Success report -> do
     winnerId <- insertPlayerIfNotExists report.winner
     loserId <- insertPlayerIfNotExists report.loser
     now <- liftIO getCurrentTime
-    insertGameReport . processReport now winnerId loserId $ report
+    insertGameReport $ processReport now winnerId loserId report
```

Comment:
The meaning of this line is identical, but I think putting the `$` there communicates the intent more clearly than the composition.